From 498599819b1078f25e1ad59598bd4f57c4b008a1 Mon Sep 17 00:00:00 2001
From: Russell King <rmk@armlinux.org.uk>
Date: Tue, 5 May 2020 01:01:01 +0300
Subject: [PATCH 15/26] pci: fix layerscape

On the LX2160A rev 1, SVR_SOC_VER() returns 0x873610, and this causes
the wrong PCIe fdt fixup code to be called, resulting in all PCIe
devices remaining disabled in the DT passed to the kernel.  Fix this
by detecting the ID and using the correct gen4 layerscape PCIe code.

Signed-off-by: Russell King <rmk@armlinux.org.uk>
---
 drivers/pci/pcie_layerscape_fixup_common.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/pcie_layerscape_fixup_common.c b/drivers/pci/pcie_layerscape_fixup_common.c
index 826f738627a5..65285c9f833f 100644
--- a/drivers/pci/pcie_layerscape_fixup_common.c
+++ b/drivers/pci/pcie_layerscape_fixup_common.c
@@ -18,8 +18,9 @@ void ft_pci_setup(void *blob, bd_t *bd)
 	uint svr;
 
 	svr = SVR_SOC_VER(get_svr());
-
-	if (svr == SVR_LX2160A && IS_SVR_REV(get_svr(), 1, 0))
+	/* Rev 1 LX2160A have svr = 0x873610 */
+	if ((svr == SVR_LX2160A || svr == (SVR_LX2160A | 0x10)) &&
+	    IS_SVR_REV(get_svr(), 1, 0))
 		ft_pci_setup_ls_gen4(blob, bd);
 	else
 #endif /* CONFIG_PCIE_LAYERSCAPE_GEN4 */
-- 
2.30.2

